File: windows-agent/src/KidGuard.Agent/Infrastructure/ApiResponse.cs
namespace KidGuard.Agent.Infrastructure;

public sealed class ApiResponse<TData>
{
    public bool Success { get; set; }

    public string Message { get; set; } = string.Empty;

    public TData? Data { get; set; }

    public List<string> Errors { get; set; } = [];
}

File: mobile/lib/src/core/api/api_response.dart
class ApiResponse<T> {
  const ApiResponse({
    required this.success,
    required this.message,
    required this.data,
    this.errors = const [],
  });

  factory ApiResponse.fromJson(
    Map<String, dynamic> json,
    T Function(Object? json) parseData,
  ) {
    final rawErrors = json['errors'];

    return ApiResponse<T>(
      success: json['success'] == true,
      message: json['message']?.toString() ?? '',
      data: parseData(json['data']),
      errors: rawErrors is List
          ? rawErrors.map((item) => item.toString()).toList()
          : const [],
    );
  }

  final bool success;
  final String message;
  final T data;
  final List<String> errors;
}
